pyccel/codegen/printing/fcode.py

```diff
@@ -1089,23 +1255,23 @@ def _print_Assign(self, expr):
         if isinstance(rhs, (PythonRange, Product)):
             return ''
 
-        if isinstance(rhs, (NumpyArray, NumpyLinspace, NumpyDiag, NumpyCross,\
-						NumpyWhere, PyccelArraySize)):
-            return rhs.fprint(self._print, expr.lhs) + '\n'
+        if isinstance(rhs, (PythonLen, NumpyRandint)):
+            rhs_code = self._print(rhs)
+            return '{0} = {1}\n'.format(lhs_code, rhs_code)
 
-        if isinstance(rhs, (NumpyFull, NumpyFullLike, NumpyEmptyLike,\
-						NumpyZerosLike, NumpyOnesLike, NumpyRand)):
+        if isinstance(rhs, PythonInt):
+            rhs = expr.rhs.fprint(self._print)
+            return '{0} = {1}\n'.format(lhs_code,rhs)
 
-            stack_array = False
-            if self._current_function:
-                name = self._current_function
-                func = self.get_function(name)
-                lhs_name = expr.lhs.name
-                vars_dict = {i.name: i for i in func.local_vars}
-                if lhs_name in vars_dict:
-                    stack_array = vars_dict[lhs_name].is_stack_array
+        if isinstance(rhs, (NumpyWhere, NumpyCross, NumpyComplex, NumpyReal, NumpyArray, NumpyLinspace)):
+            rhs = self._print(rhs)
+            return '{0} = {1}\n'.format(lhs_code,rhs)
```

Comment:
This block isn't needed. The same calls occur if you remove this code

Reply:
See:
https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/printing/fcode.py#L1169-L1177